src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1542,14 +1580,17 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                         filterOpt: null,
                         cancellationToken: cancellationToken);
 
-                    if (success)
+                    if (!moduleBeingBuilt.EmitOptions.EmitMetadataOnly)
                     {
-                        success = GenerateResourcesAndDocumentationComments(
+                        if (!GenerateResourcesAndDocumentationComments(
```

Comment:
This matches the existing behavior in future: see `CSharpCompilation.CompileImpl`.